player-vs-computer.js

```diff
@@ -0,0 +1,240 @@
+    return;
+  };
+
+  // ensure range top is valid
+  async function checkTopRange(input) {
+    if (!isNumeric(input)) {
+      console.log(
+        `\n${input} is not a valid number, ${name}. Enter something else please!\n`
+      );
+      await getTopRange();
+    } else if (isNumeric(input) === true && parseFloat(input) % 1 != 0) {
+      console.log("\nIntegers only, please!\n");
+      await getTopRange();
+    } else if (isNumeric(input) === true && parseInt(input) < 0) {
+      console.log("\nNo negative numbers, please. Let's try again.\n");
+      await getTopRange();
+    } else if (parseInt(input) < rangeBottom) {
+      console.log(
+        "\nThe top end of the range has to be higher than the low end. We both know you know that.\n"
+      );
+      await getTopRange();
+    } else {
+      return;
+    };
+  };
+
+  // function to prevent user from entering anything other than a "numerical string", e.g. "dghdfgd"
+  function isNumeric(num) {
+    return !isNaN(parseFloat(num)) && isFinite(num);
+  };
+
+  //generate computer's 'secret number'
+  let computerNum = randomNum(rangeBottom, rangeTop);
+
+  //generate computer's guess
+  let computerGuess = binarySearch(rangeBottom, rangeTop);
+  console.log("\nOk, I have selected my secret number out of that range.\n");
+
+  await getPlayerNum();
+
+  async function getPlayerNum() {
+    playerNum = await ask(
+      "Now, what is your secret number?\nI won't peek, I promise...\n\n"
+    );
+    await checkPlayerNum(playerNum);
+  };
+
+  async function checkPlayerNum(input) {
+    if (!isNumeric(input)) {
+      console.log(
+        `${input} is not a valid number, ${name}. Enter something else please!\n\n`
+      );
+      await getPlayerNum();
+    } else if (isNumeric(input) === true && parseFloat(input) % 1 != 0) {
+      console.log("\nIntegers only, please!\n");
+      await getPlayerNum();
+    } else if (parseInt(input) < rangeBottom || parseInt(input) > rangeTop) {
+      console.log(
+        "\nThe command line wizard told me your number is outside of the range we agreed upon. That is deeply unchill of you. Please pick a new number within the range.\n"
+      );
+      await getPlayerNum();
+    };
+  };
+
+  // print player's choice for accountability (no cheating!)
+  console.log("\nYou entered: " + playerNum);
+
+  // player guesses first
+  console.log(
+    `\nOk, let\'s begin. You can guess first. Go ahead and guess a number between ${rangeBottom} and ${rangeTop}.\n`
+  );
+
+  playerTurn();
+
+  // all logic for player turn
+  async function playerTurn() {
+    let guessInput = await ask("Guess!\n\n");
+    let guess = parseInt(guessInput);
+    if (guess === computerNum) {
+      console.log("\nThat's it! You guessed the number!\n");
+      process.exit();
+    } else if (guess < computerNum) {
+      console.log("\nThat's not it. My number is higher.");
+    } else if (guess > computerNum) {
+      console.log("\nThat's not it. My number is lower.");
+    }
```

Comment:
This is the only spot that doesn't have checks to be sure that the user inputs are proper things to put in.